cmd/ipfs/ulimit_unix.go

```diff
@@ -3,40 +3,50 @@
 package main
 
 import (
+	"errors"
 	"fmt"
 	"syscall"
 )
 
-func init() {
-	fileDescriptorCheck = checkAndSetUlimit
-}
-
-func checkAndSetUlimit() error {
-	var rLimit syscall.Rlimit
-	err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rLimit)
-	if err != nil {
+// checks for the maximum number of file descriptors that the system supports
+// if we have a smaller softer, maximum value than the IPFS_FD_MAX
+// current env variable then we shall overwrite them with it
+func managefd() (err error) {
+	rlimit := syscall.Rlimit{}
+	if err = syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rlimit); err != nil {
 		return fmt.Errorf("error getting rlimit: %s", err)
 	}
 
-	var setting bool
-	if rLimit.Cur < ipfsFileDescNum {
-		if rLimit.Max < ipfsFileDescNum {
-			log.Error("adjusting max")
-			rLimit.Max = ipfsFileDescNum
+	max := rlimit.Max
+	rlimit.Cur, rlimit.Max = maxFds, maxFds
+
+	// the soft limit is the value that the kernel enforces for the corresponding resource
+	// the hard limit acts as a ceiling for the soft limit
+	// an unprivileged process may only set its soft limit to a value
+	// in the range from 0 up to the hard limit
+	if err = syscall.Setrlimit(syscall.RLIMIT_NOFILE, &rlimit); err != nil {
```

Comment:
This logic is nearly identical to the other freebsd logic. I'm sure we can DRY it up a bit